src/common/btree.c

```diff
@@ -4183,8 +4183,12 @@ dbtree_class_register(unsigned int tree_class, uint64_t tree_feats,
 		return -DER_INVAL;
 
 	/* XXX should be multi-thread safe */
-	if (btr_class_registered[tree_class].tc_ops != NULL)
-		return -DER_EXIST;
+	if (btr_class_registered[tree_class].tc_ops != NULL) {
+		if (btr_class_registered[tree_class].tc_ops != ops ||
+		    btr_class_registered[tree_class].tc_feats != tree_feats)
+			return -DER_EXIST;
+		return 0;
```

Comment:
This change looks good to me, but I don't see why repeated vos_self_init() could fail with -DER_EXIST, the -DER_EXIST errors should have been ignored by most dbtree_class_register() callers, including the vos_self_init().

Reply:
I didn't see that it was ignored. Would you prefer that I change it so that each call of dbtree_class_register() ignores the DER_EXIST error instead of handling it within dbtree_class_register()?

Reply:
oh, I was looking into vos_self_init() -> vos_self_nvme_init(), but seems vos_self_init() -> vos_mod_init() does return error in such case.
I'm fine with current approach.